pennylane/plugins/default_qubit.py

```diff
@@ -302,11 +302,11 @@ def apply(self, operation, wires, par):
                 raise ValueError('State vector must be of length 2**wires.')
             return
         elif operation == 'BasisState':
+            n = len(par[0])
             # get computational basis state number
-            if not (set(par[0]) == {0, 1} or set(par[0]) == {0} or set(par[0]) == {1}):
-                raise ValueError("BasisState parameter must be an array of 0/1 integers.")
+            if not (set(par[0]) == {0, 1} or set(par[0]) == {0} or set(par[0]) == {1}) or (wires != None and wires != [] and n != len(wires)):
+                raise ValueError("BasisState parameter must be an array of len(wires) many 0/1 integers."+str(n)+" "+str(wires))
```

Comment:
Won't this lead to strings like:
"BasisState parameter must be an array of len(wires) many 0/1 integers.2 2"
?

Reply:
Perhaps the following?
```python3
"BasisState parameter must be an length-{} array of 0 or 1 integers.".format(2**len(wires))
```

Reply:
I will use "be a length-{}"